hack/chains/kaniko-demo.sh

```diff
@@ -21,17 +21,29 @@ oc project tekton-chains
 title "Ensure we have the demo kaniko build task"
 kubectl apply -f https://raw.githubusercontent.com/tektoncd/chains/main/examples/kaniko/kaniko.yaml
 
-# Tweak it to avoid TLS failures
-# Fixme: Make it work without --skip-tls-verify
-kubectl patch task/kaniko-chains --type=json \
-  --patch='[{"op": "add", "path": "/spec/steps/1/args/5", "value":"--skip-tls-verify=true"}]'
-
 IMAGE_REGISTRY=$( oc registry info )
 DOCKERCFG=$( oc get sa tekton-chains-controller -o json | jq -r '.secrets[1].name' )
 
+# When accessing the registry via ingress/route the certificate issued for the registry endpoint
+# is not mounted via /var/run/secrets so not picked up by kaniko, here we add the ingress CA
+# certificate to the $DOCKERCFG secret and pass it via parameter to kaniko
+EXTRA_ARGS=""
+if [[  $IMAGE_REGISTRY != 'image-registry.openshift-image-registry.svc:5000' ]]; then
+  title "Setting up CA trust with exported registry"
+  # Fetch the ingress CA certificate
+  INGRESS_CA=$(oc get secret router-ca -n openshift-ingress-operator -o jsonpath="{.data.tls\.crt}" )
+  # create configmap with the ingress CA certificate
+  oc create configmap ingress-ca.crt --from-file=ingress-ca.crt=<(echo $INGRESS_CA) 2>&1 | grep -q 'already exists' \
```

Comment:
It might not matter, but if $INGRESS_CA could have line breaks in it, then perhaps it's better to put quotes in `echo "$INGRESS_CA"`